tap_salesforce/__init__.py

```diff
@@ -302,13 +303,16 @@ def do_discover_object(sf):
     unsupported_fields = set()
     properties = {}
     mdata = metadata.new()
+    source_column_types = {}
```

Comment:
Some Salesforce fields (e.g. date, int) need be formatted specifically in the filter, or we'll get an error. Since we translate the SF types to our more general Symon types, I'm keeping track of the original types here and adding it to the catalog, so if a filter is applied to columns of these types, we can change the format of the filter value appropriately.